### PR TITLE
ROX-25748: Workload CVE a11y fixes

### PR DESCRIPTION
**Pull request by dvail**
### Problem reported by axe DevTools

> Ensure ARIA attributes are not prohibited for an element's role

https://dequeuniversity.com/rules/axe/4.10/aria-prohibited-attr?application=AxeChrome2

<div class="pf-v5-c-toolbar__group pf-v5-u-w-100" aria-label="applied search filters"><div class="pf-v5-l-flex pf-m-space-items-xs search-filter-chips"></div></div>

### Analysis

> aria-label attribute cannot be used on a div with no valid role attribute.

### Solution

The offending `div` has no valid role because it is an empty element when no filters are applied. To fix, conditionally render this element only when there are applied filters and valid children.

Bonus! This removes a little bit of extra whitespace in the layout due to the empty element's padding.

---

### Problem reported by axe DevTools

> Ensure all ARIA attributes have valid values

https://dequeuniversity.com/rules/axe/4.10/aria-valid-attr-value?application=AxeChrome

<button type="button" data-ouia-component-type="PF5/TabButton" data-ouia-safe="true" class="pf-v5-c-tabs__link" id="pf-tab-OBSERVED-pf-1723741011927z7xby7wrjx8" aria-controls="pf-tab-section-OBSERVED-pf-1723741011927z7xby7wrjx8" role="tab" aria-selected="true">

### Analysis

> Invalid ARIA attribute value: aria-controls="pf-tab-section-OBSERVED-pf-1723741011927z7xby7wrjx8"

### Solution

The `aria-controls` attribute is invalid because it does not correspond to the `id` attribute of any other element in the DOM. The autogenerated ID does not match another element because these `<Tab>` components are disconnected from their content. To fix, a static value was passed to the `tabContentId` props and used as the `id` field of the element on the page that is controlled by the tab navigation.

---

### Problem reported by axe DevTools

> Ensure all ARIA attributes have valid values

https://dequeuniversity.com/rules/axe/4.10/button-name?application=AxeChrome

<button id="copy-image-name-button-sha256:8b03e4185ecd305bc9b410faac15d486a3b1ef1946196d429245cdd3c7b152eb" aria-labelledby="copy-image-name-text-sha256:8b03e4185ecd305bc9b410faac15d486a3b1ef1946196d429245cdd3c7b152eb" aria-disabled="false" class="pf-v5-c-button pf-m-plain pf-v5-u-pt-xs" type="button" data-ouia-component-type="PF5/Button" data-ouia-safe="true" data-ouia-component-id="OUIA-Generated-Button-plain-6">

### Analysis

> aria-label attribute does not exist or is empty
> aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty

### Solution

The `aria-labelledby` attribute is set for these copy buttons, but is never set as the id of an element that can be used as a label. Since there is no element that consistently displays a label value that is appropriate for this button, the attribute was changed to `aria-label="Copy image name"`.

### User-facing documentation

- [ ] CHANGELOG is updated **OR** update is not needed
- [ ] [documentation PR](https://spaces.redhat.com/display/StackRox/Submitting+a+User+Documentation+Pull+Request) is created and is linked above **OR** is not needed

### Testing and quality

- [ ] the change is production ready: the change is GA or otherwise the functionality is gated by a feature flag
- [ ] CI results are inspected

#### Automated testing

- [ ] added unit tests
- [ ] added e2e tests
- [ ] added regression tests
- [ ] added compatibility tests
- [ ] modified existing tests

#### How I validated my change

Visit the Workload CVE Overview page => Image tab and see the presence of 16 errors due to the three rule violations:
image

Visit the same page after the changes are applied and see zero reported rule violations:
image

Repeat these checks across the following pages, verifying that the axe errors above are not present.

Workload CVE Overview page - CVE, Image and Deployment tabs
Workload CVE page - Image and Deployment tabs
Workload CVE Image page
Workload CVE Deployment page

### CODE REVIEW

File: ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/components/VulnerabilityStateTabs.tsx [hunk 1]
```diff
@@ -42,10 +44,12 @@ function VulnerabilityStateTabs({
             <Tab
                 eventKey="OBSERVED"
                 title={<TabTitleText>{titleOverrides?.observed ?? 'Observed'}</TabTitleText>}
+                tabContentId={vulnStateTabContentId}
```
Comment (sachaudh) [inline ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/components/VulnerabilityStateTabs.tsx:47]: I was a little confused about this one. Should the `tabContentId` for each be unique?

**Review by sachaudh — APPROVED** [review]

File: ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/components/VulnerabilityStateTabs.tsx (hunk 1 above)
Comment (dvail) [reply ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/components/VulnerabilityStateTabs.tsx:47]: In the case of each tab having it's own target container, I would make each one unique to that container. In this case though, each tab drives the content for just a single container element so we don't have three different values we can use.

Comment (pedrottimark) [reply ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/components/VulnerabilityStateTabs.tsx:47]: Somewhat similar to need for only one `id` in #12238

**Review by pedrottimark — APPROVED** [review]

**Pull request merged**